pkg/installmanager/installmanager.go

```diff
@@ -1803,10 +1807,93 @@ func (m *InstallManager) overrideCredentialsModeOnInstallConfig(credentialsMode
 	}
 
 	if err := ioutil.WriteFile(clusterConfigPath, clusterConfigBytes, 0644); err != nil {
-		return errors.Wrap(err, "error writing install-config.yaml")
+		return errors.Wrap(err, "error writing cluster-config.yaml")
 	}
 
 	m.log.WithField("newMode", credentialsMode).Info("overrode effective in-cluster credentials mode")
 
+	if cd.Spec.Platform.Azure == nil {
+		return nil
+	}
+
+	//If Azure, need fill up azure_resource_prefix and azure_resourcegroup for 99_cloud-creds-secret.yaml
+	m.log.Info("patch 99_cloud-creds-secret.yaml when platform is azure")
+	region := installConfig.Platform.Azure.Region
```

Comment:
At first glance, I couldn't determine that this field would always be set; but I found that we have [webhook validation](https://github.com/openshift/hive/blob/b6d323da70b09053352d87f8ec47ff2a5be2992e/pkg/validating-webhooks/hive/v1/clusterdeployment_validating_admission_hook.go#L479-L481) to make sure it is, so this is okay.

Reply:
ok